crates/primitives-traits/src/block/header.rs

```diff
@@ -35,5 +35,24 @@ pub trait BlockHeader:
     + 'static
 {
 }
-
-impl BlockHeader for alloy_consensus::Header {}
+impl<T> BlockHeader for T where
+    T: Send
+        + Sync
+        + Unpin
+        + Clone
+        + Hash
+        + Default
+        + fmt::Debug
+        + PartialEq
+        + Eq
+        + alloy_rlp::Encodable
+        + alloy_rlp::Decodable
+        + alloy_consensus::BlockHeader
+        + Sealable
+        + InMemorySize
+        + MaybeSerde
+        + MaybeSerdeBincodeCompat
+        + AsRef<T>
+        + 'static
+{
+}
```

Comment:
this makes sense but maybe isn't strictly necessary because I think implementing the trait like 

impl BlockHeader for alloy_consensus::Header {}

isn't a big deal

Reply:
absolutely, done here https://github.com/paradigmxyz/reth/pull/13995/commits/b04b2f4fa725d51720eb53396029d207779489a4